src/sentry/preprod/api/endpoints/organization_preprod_artifact_assemble.py

```diff
@@ -0,0 +1,124 @@
+                "chunks": "The chunks field is required and must be provided as an array of 40-character hexadecimal strings.",
+                "file_name": "The file_name field must be a string.",
+                "sha": "The sha field must be a string.",
+                "build_configuration": "The build_configuration field must be a string.",
+                "extras": "The extras field must be an object.",
+            }
+
+            try:
+                data = orjson.loads(request.body)
+                jsonschema.validate(data, schema)
+            except jsonschema.ValidationError as e:
+                error_message = e.message
+                # Get the field from the path if available
+                if e.path:
+                    if field := e.path[0]:
+                        error_message = error_messages.get(str(field), error_message)
+
+                return Response({"error": error_message}, status=400)
+            except Exception:
+                return Response({"error": "Invalid json body"}, status=400)
+
+            checksum = data.get("checksum")
+            chunks = data.get("chunks", [])
+
+            # Check if all requested chunks have been uploaded
+            missing_chunks = find_missing_chunks(project.organization_id, set(chunks))
+            if missing_chunks:
+                return Response(
+                    {
+                        "state": ChunkFileState.NOT_FOUND,
+                        "missingChunks": missing_chunks,
+                    }
+                )
+
+            # Check current assembly status
+            state, detail = get_assemble_status(AssembleTask.PREPROD_ARTIFACT, project.id, checksum)
+            if state == ChunkFileState.OK:
+                return Response({"state": state, "detail": detail, "missingChunks": []})
+            elif state is not None:
+                return Response({"state": state, "detail": detail, "missingChunks": []})
+
+            # There is neither a known file nor a cached state, so we will
+            # have to create a new file.  Assure that there are checksums.
+            # If not, we assume this is a poll and report NOT_FOUND
+            if not chunks:
+                return Response({"state": ChunkFileState.NOT_FOUND, "missingChunks": []})
+
+            set_assemble_status(
+                AssembleTask.PREPROD_ARTIFACT, project.id, checksum, ChunkFileState.CREATED
+            )
+
+            # Trigger assembly task
+            from sentry.tasks.assemble import assemble_preprod_artifact
+
+            assemble_preprod_artifact.apply_async(
+                kwargs={
+                    "org_id": project.organization_id,
+                    "project_id": project.id,
+                    "checksum": checksum,
+                    "chunks": chunks,
+                    "file_name": data.get("file_name"),
+                    "sha": data.get("sha"),
```

Comment:
nit: since this assemble api is doing some checksum stuff, maybe this should be `git_sha`?